ui/app/utils/classes/exec-command-editor-xterm-adapter.js

```diff
@@ -0,0 +1,40 @@
+const REVERSE_WRAPAROUND_MODE = '\x1b[?45h';
+
+export default class ExecCommandEditorXtermAdapter {
+  constructor(terminal, setCommandCallback, command) {
+    this.terminal = terminal;
+    this.setCommandCallback = setCommandCallback;
+
+    this.command = command;
+
+    this.keyListener = terminal.onKey(e => {
+      this.handleKeyEvent(e);
+    });
+
+    // Allows tests to bypass synthetic keyboard event restrictions
+    terminal.simulateCommandKeyEvent = this.handleKeyEvent.bind(this);
+
+    terminal.write(REVERSE_WRAPAROUND_MODE);
+  }
+
+  handleKeyEvent(e) {
+    // Issue to handle arrow keys etc: https://github.com/hashicorp/nomad/issues/7463
+    if (e.domEvent.key === 'Enter') {
+      this.terminal.writeln('');
+      this.setCommandCallback(this.command);
+      this.keyListener.dispose();
+    } else if (e.domEvent.key === 'Backspace') {
+      if (this.command.length > 0) {
+        this.terminal.write('\x08 \x08');
```

Comment:
I think the `\x08` incantation is worth sticking in a constant like `REVERSE_WRAPAROUND_MODE`.

Reply:
Agreed, thank you, I’ve added a commit with that.